### PR TITLE
[gpkg] Ensure that tables present in gpkgext_relations can be read

### PR DESCRIPTION
**Pull request by nyalldawson**
Tables which are listed in gpkgext_relations but NOT gpkg_contents should be readable as valid layers. The GeoPackage related tables extension explicitly states that tables in gpkgext_relations are not required to be listed in gpkg_contents.

### CODE REVIEW

File: autotest/ogr/ogr_gpkg.py [hunk 1]
```diff
@@ -7269,6 +7276,21 @@ def get_query_row_count(query):
         )
         == 1
     )
+    # force delete from gpkg_contents, and then ensure that we CAN successfully
+    # load layers which are present ONLY in gpkgext_relations but NOT
+    # gpkg_contents (i.e. datasources which fully the Related Tables specification
```
Comment (rouault) [inline autotest/ogr/ogr_gpkg.py:7281]:
```suggestion
    # gpkg_contents (i.e. datasources which follow the Related Tables specification
```

**Review by rouault — COMMENTED** [review]
the approach seems very reasonable to me

File: ogr/ogrsf_frmts/gpkg/ogrgeopackagedatasource.cpp [hunk 2]
```diff
@@ -1575,6 +1576,12 @@ int GDALGeoPackageDataset::Open(GDALOpenInfo *poOpenInfo,
                           "name = 'gpkg_geometry_columns' AND "
                           "type IN ('table', 'view')",
                           nullptr) == 1;
+        bHasGPKGExtRelations =
+            SQLGetInteger(hDB,
+                          "SELECT 1 FROM sqlite_master WHERE "
+                          "name = 'gpkgext_relations' AND "
+                          "type IN ('table', 'view')",
+                          nullptr) == 1;
```
Comment (rouault) [inline ogr/ogrsf_frmts/gpkg/ogrgeopackagedatasource.cpp:1584]:
```suggestion
        bHasGPKGExtRelations = HasGpkgextRelationsTable();
```

Comment (nyalldawson) [reply ogr/ogrsf_frmts/gpkg/ogrgeopackagedatasource.cpp:1584]: Done

**Pull request merged**